Sources/TuistKit/Services/TestService.swift

```diff
@@ -571,8 +578,15 @@ final class TestService { // swiftlint:disable:this type_body_length
                     .first(
                         where: { $0.name == testPlanConfiguration.testPlan }
                     )?.testTargets.map(\.target) ?? []
+            } else if let testActionTargets = scheme.testAction?.targets.map(\.target), !testActionTargets.isEmpty {
+                testActionTargets
+            } else if let testPlanTargets = scheme.testAction?.testPlans?.flatMap({ $0.testTargets.map(\.target) }),
+                      !testPlanTargets.isEmpty
+            {
+                // If no test plan is specified, we use the targets of _all_ test plans
```

Comment:
Should we update this to respect which targets are actually run based on the default test plan (if such exists)?

Reply:
Done 👍